tools/src/tester/TestsRunner.ts

```diff
@@ -0,0 +1,65 @@
+import { type OpenAPIV3 } from 'openapi-types'
+import SpecParser from './SpecParser'
+import ChapterReader from './ChapterReader'
+import SchemaValidator from './SchemaValidator'
+import StoryEvaluator, { type StoryFile } from './StoryEvaluator'
+import fs from 'fs'
+import { type Story } from './types/story.types'
+import { read_yaml } from '../../helpers'
+import { type StoryEvaluation } from './types/eval.types'
+import ResultsDisplayer from './ResultsDisplayer'
+
+declare global {
+  // eslint-disable-next-line no-var
+  var chapter_reader: ChapterReader
+  // eslint-disable-next-line no-var
+  var schema_validator: SchemaValidator
+  // eslint-disable-next-line no-var
+  var spec_parser: SpecParser
+}
+
+export default class TestsRunner {
+  path: string // Path to a story file or a directory containing story files
+
+  constructor (spec: OpenAPIV3.Document, path: string) {
+    // TODO: Grab server URL from environment variable and add authentication.
+    global.chapter_reader = new ChapterReader('http://localhost:9200')
+    global.spec_parser = new SpecParser(spec)
+    global.schema_validator = new SchemaValidator(spec)
+    this.path = path
+  }
+
+  async run (ignore_skipped = false, ignore_passed = true): Promise<void> {
+    const evaluations = await this.evaluate()
+    for (const evaluation of evaluations) {
+      const displayer = new ResultsDisplayer(evaluation)
+      displayer.display(ignore_skipped, ignore_passed)
+    }
+  }
+
+  async evaluate (): Promise<StoryEvaluation[]> {
+    const story_files = this.#collect_story_files(`${process.cwd()}/${this.path}`, '', '')
```

Comment:
This would be safer to use [`path.resolve(this.path)`](https://nodejs.org/api/path.html#pathresolvepaths) so that it handles the case where `this.path` might already be an absolute path